### DISCUSSION TITLE
Strange Datatable Issue

### DISCUSSION THREAD

**Discussion by suaveone** [livewire/livewire#3202] [7. Bugs (archived)] [answered]
I have a data-table that works perfectly on my local dev environment. It filters through a paginated set of 11k records when user selects a filter type. Not sure if its a code issue or server issue...

When I load the exact same code on Forge Servers, 4 core with 8G memory SSD, the data-table rows do not show in the list for all records, and larger found sets. However I see the pagination counts totals up correctly and the pagination buttons all working, but the rows do not display??

Very Strange.  I noticed that filters that have large sets, like "All Cases" which should return all 11k records, don't show records, but smaller sets under 500 show... Again all the code works perfect on my local dev. If anyone has seen this would help tremendously since my client wants to see a demo this week.

https://user-images.githubusercontent.com/30648644/123693499-01204a80-d80d-11eb-92d4-56ca7c0d09b1.mov

https://user-images.githubusercontent.com/30648644/123693546-0aa9b280-d80d-11eb-80a4-4f2012f93199.mov

**Comment by joshhanley** [comment, marked as answer]
@suaveone on the first server where you aren't getting any data in the big tables, do you have `APP_DEBUG=false` by any chance as it's a production server? Maybe you are getting the sort memory error on both servers, but the second one is displaying it due to debugging turned on?

If you can confirm if both are getting the sort buffer issue or not, then that should help narrow down what you will need to fix the issue.

I suspect it's the same issue on both. I would be investigating how your query runs it's sorts or considering increasing the sort buffer in the mysql config. Seems there might be a bug in mysql if you are trying to sort by JSON columns, see here https://stackoverflow.com/a/67266529/14909455

Hope this helps!

**Reply by suaveone** [reply]
Thanks.  I put the server production but left debug=true.  I also tried some keyword searches like "Eric" and the pagnation count gives the following but still no rows showing.  I'm stumped.
![Screen Shot 2021-06-28 at 9 36 09 PM](https://user-images.githubusercontent.com/30648644/123738213-3bafd480-d859-11eb-9106-011353b31c57.png)

**Reply by joshhanley** [reply]
@suaveone hmm how odd.

Can you put a dump in there to see if your results/ case archives property is actually getting data? Also turn on db logging to see if the query was run and whether any results were returned from the DB. I suspect it's a DB issue on your servers.

**Reply by suaveone** [reply]
Thanks the the advice.  I was not thinking it was a Database Issue since the queries were working, however based on your advice I found that it was! Some columns that were used in the joins were not indexed.  So the database was using its memory to buffer, as opposed to my virtual local dev environment that leveraged my local ram, so it never overloaded the buffer.  Thanks so much the issue has been fixed.

**Reply by joshhanley** [reply]
@suaveone no worries! Glad it helped sort it out 🙂
